internal/api/courses/assignments/submissions/proxy/regrade.go

```diff
@@ -0,0 +1,60 @@
+package proxy
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/model"
+)
+
+type RegradeRequest struct {
+	core.APIRequestAssignmentContext
+	core.MinCourseRoleGrader
+
+	// Filter results to only users with this role.
+	FilterRole model.CourseUserRole `json:"filter-role"`
+
+	// Wait for the entire regrade to complete and return all results.
+	WaitForCompletion bool `json:"wait-for-completion"`
+}
+
+type RegradeResponse struct {
+	Complete bool                                    `json:"complete"`
```

Comment:
I don't think this value really makes sense without caching.
For analysis, it can make sense because you may not be waiting, but get all your results from the cache.